include/vkwrap/vkwrap/image.h

```diff
@@ -0,0 +1,316 @@
+
+        view_create_info.setImage( image );
+        view_create_info.setViewType( detail::chooseImageViewType( image_create_info.imageType ) );
+        view_create_info.setFormat( image_create_info.format );
+
+        view_create_info.subresourceRange.setAspectMask( chooseAspectMask( image_create_info.format ) );
+        view_create_info.subresourceRange.setLayerCount( image_create_info.arrayLayers );
+
+        return view_create_info;
+    } // makeImageViewCreateInfo
+
+    static vk::UniqueImageView createImageView( const vk::ImageViewCreateInfo& create_info, vk::Device device )
+    {
+        return device.createImageViewUnique( create_info );
+    } // createImageView
+
+    friend void swap( Image& lhs, Image& rhs )
+    {
+        std::swap( lhs.get(), rhs.get() );
+        std::swap( lhs.m_mman, rhs.m_mman );
+        std::swap( lhs.m_view, rhs.m_view );
+    } // swap
+
+  public:
+    Image( const vk::ImageCreateInfo& create_info, Mman& mman )
+        : Base{ createImage( create_info, mman ) },
+          m_mman{ &mman },
+          m_view{ createImageView( makeImageViewCreateInfo( create_info, *this ), mman.getDevice() ) } //
+    {
+    } // Image
+
+    ~Image()
+    {
+        /** There cannot be an exception.
+         *
+         * 1. If `createImage` throws an exception in constructor,
+         * then `m_mman == nullptr` and we won't `destroy` an object.
+         *
+         * 2. If constructor completes successfully,
+         * then `Mman::findInfo` CANNOT throw an exception:
+         * - info must be found in `Mman::m_images_info`
+         * - as `vk::Image` and `Mman::ImageInfo` have only noexcept methods,
+         *   then `unordered_map::erase` cannot throw an exception
+         */
+        if ( m_mman != nullptr )
+        {
+            m_mman->destroy( *this );
+        }
+    } // ~Image
+
+    Image( Image&& image )
+        : Base{ nullptr },
+          m_mman{ nullptr }
+    {
+        swap( *this, image );
+    } // Image( Image&& )
+
+    Image& operator=( Image&& image )
+    {
+        Image tmp{ std::move( image ) };
+        swap( *this, tmp );
+
+        return *this;
+    } // operator=( Image&& )
+
+    Image( const Image& ) = delete;
+    Image& operator=( const Image& ) = delete;
+
+    void update( vk::Buffer src_buffer ) { m_mman->copy( src_buffer, *this ); }
+    void update( vk::Buffer src_buffer, Mman::RegionMaker maker )
+    {
+        m_mman->copy( src_buffer, *this, maker );
+    } // update
+
+    void transit( vk::ImageLayout new_layout ) { m_mman->transit( *this, new_layout ); }
+
+    vk::ImageView getView() const { return m_view.get(); }
+
+    Base& get() { return static_cast<Base&>( *this ); }
+    const Base& get() const { return static_cast<const Base&>( *this ); }
+
+    using Base::operator bool;
+    using Base::operator VkImage;
+
+}; // class Image
+
+class ImageBuilder
+{
+
+  public:
+    using Queues = std::vector<Queue>;
```

Comment:
This alias should go into the vkwrap/queues.h

Reply:
This alias is the way to represent range of Queue in that wrapper. Actually, I need to make private alias.